### PR TITLE
TLS fixes

### PR DESCRIPTION
**Pull request by bergundy**
## What
See commit messages
## Why?
- Uniformity with other SDKs
- Some recent features were broken

### CODE REVIEW

File: packages/client/src/index.ts [hunk 1]
```diff
@@ -359,11 +362,27 @@ export function compileWorkflowOptions({
 }
 
 /**
- * Convert {@link ConnectionOptions.tls} to {@link grpc.ChannelCredentials} and
- * add the grpc.ssl_target_name_override GRPC {@link ConnectionOptions.channelArgs}
+ * Normalize {@link ConnectionOptions.tls} by turning false and null to undefined and true to and empty object
+ * NOTE: this function is duplicated in `packages/worker/src/worker.ts` for lack of a shared library
+ */
+function normalizeTlsConfig(tls?: ConnectionOptions['tls']): TLSConfig | undefined {
+  return typeof tls === 'object' ? (tls === null ? undefined : tls) : tls ? {} : undefined;
```
Comment (Sushisource) [inline packages/client/src/index.ts:369]: This is so painfully javascript :laughing:

File: packages/worker/native/src/lib.rs [hunk 2]
```diff
@@ -442,46 +439,47 @@ fn worker_new(mut cx: FunctionContext) -> JsResult<JsUndefined> {
         target_url: Url::parse(&url).unwrap(),
         namespace: server_options
             .get(&mut cx, "namespace")?
-            .downcast_or_throw::<JsString, FunctionContext>(&mut cx)?
+            .downcast_or_throw::<JsString, _>(&mut cx)?
             .value(&mut cx),
```
Comment (Sushisource) [inline packages/worker/native/src/lib.rs:443]: This 3 line preamble is a bit visually noisy. What do you think about replacing it with this (written off my head, not guaranteed to compile :smile: )

```rust
macro_rules! js_value {
    ($js_obj:ident, $js_cx:ident, $prop_name:expr, $js_type:ty) => {
        $js_obj
            .get(&mut $js_cx, $prop_name)?
            .downcast_or_throw::<$js_type, _>(&mut $js_cx)?
            .value(&mut $js_cx)
    }
}
```

Then you can just say `js_value!(server_options, cx, "namespace", JsString)`

File: packages/worker/native/index.d.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// NOTE: this interface is duplicated in the client module `packages/client/src/index.ts`
+// NOTE: this interface is duplicated in the client module `packages/client/src/index.ts` for lack of a shared library
```
Comment (Sushisource) [inline packages/worker/native/index.d.ts:1]: Any reason not to just make a lower dep they can share?

Comment (bergundy) [reply packages/worker/native/index.d.ts:1]: I will at some point, I just didn't want to add to this PR.

File: packages/worker/native/src/lib.rs (hunk 2 above)
Comment (bergundy) [reply packages/worker/native/src/lib.rs:443]: I'll leave that for a subsequent PR, good idea, thanks.

File: packages/client/src/index.ts (hunk 1 above)
Comment (bergundy) [reply packages/client/src/index.ts:369]: Yup ..

**Review by bergundy — COMMENTED** [review]
Thanks for the input @Sushisource.

**Comment by bergundy** [comment]
@Sushisource I've pushed 2 new commits, we're now testing mTLS against a Temporal Cloud deployment.
Note that I've handled an error in the bridge loop where there once was a TODO, I've reached it when starting a worker without polling and waiting for the process to exit (the worker was dropped while the loop was running).

**Pull request merged**